packages/model-viewer/src/three-components/ARRenderer.ts

```diff
@@ -176,6 +207,11 @@ export class ARRenderer extends EventDispatcher<
       console.warn('Cannot present while a model is already presenting');
     }
 
+    scene.add(this.controller1);
+    scene.add(this.controller2);
+    scene.add(this.controllerGrip1);
+    scene.add(this.controllerGrip2);
```

Comment:
Regarding controllers of the wrong size, you can see further down this file that scene.scale is changed when the object is scaled by pinch on a phone. We may need to add another level of object hierarchy to the AR scene to fix this. However, I believe that scale starts at 1, so if you're seeing a problem from the beginning, it may be a different issue. Can you add a screen recording by chance?

Reply:
Okay, I just saw your video on the other thread, thanks. Interesting, that does seem like a different problem than scene scale. Position maybe? Let's move discussion from the thread to this PR, FYI @cabanier.

Reply:
Yes, it's a position problem, not a scale problem. The controllers seems small because they are far away from the user. E.g. I've found that initially the controller model is in the right place, but then the 3D model drops to the floor and the controller gets lower.

Changing the `isFirstView` variable to `false`:

```ts
// WebXR may return multiple views, i.e. for headset AR. This
// isn't really supported at this point, but make a best-effort
// attempt to render other views also, using the first view
// as the main viewpoint.
let isFirstView: boolean = true;
for (const view of pose.views) {
  this.updateView(view);

  if (isFirstView) {
    this.moveToFloor(frame);

    this.processInput(frame);

    const delta = time - this.lastTick!;
    this.moveScene(delta);
    this.renderer.preRender(scene, time, delta);
    this.lastTick = time;

    scene.renderShadow(this.threeRenderer);
  }

  this.threeRenderer.render(scene, scene.getCamera());
  isFirstView = false;
}
```

and commenting/removing the following code:

```ts
const {theta, radius} =
    (element as ModelViewerElementBase & ControlsInterface)
        .getCameraOrbit();
// Orient model to match the 3D camera view
const cameraDirection = xrCamera.getWorldDirection(vector3);
scene.yaw = Math.atan2(-cameraDirection.x, -cameraDirection.z) - theta;
this.goalYaw = scene.yaw;

position.copy(xrCamera.position)
    .add(cameraDirection.multiplyScalar(radius));

this.updateTarget();
const target = scene.getTarget();
position.add(target).sub(this.oldTarget);

this.goalPosition.copy(position);
```

fixes the problem.

Reply:
Do you understand why this is? Any recommendation for a better approach?

Reply:
Controllers are added to the scene. The position of the scene changes, the controllers move with it.

Reply:
Okay, then it sounds like adding an extra node in the scene hierarchy is the way to go.

Reply:
Could you please help with that? 🙏 And I'll focus on other aspects and bugs.

Reply:
@elalish Any progress?

Reply:
Sadly no, I've been slammed. We did finally get a Quest 3, but no one's even had time to open the box yet...

Reply:
Hi @FluorescentHallucinogen, amazing work for the Google modelviewer! I will try to test it on my Quest 3 some time.

Reply:
Yesterday I tried getting it working, but no luck. Any way to build the model-viewer module from your fork or can I test already?

Reply:
@MickGerrit Hi! Thanks!

The only difference in my fork is in one single `/packages/model-viewer/src/three-components/ARRenderer.ts` file, everything else is the same as the original project. Have you tried to build the original project as described in https://github.com/google/model-viewer?tab=readme-ov-file#development?

Reply:
@MickGerrit, I've deployed 3 demos for you. :wink:

https://model-viewer-xr-drag.web.app/fixed/ — it uses the code from this pull request + the hack described in the https://github.com/google/model-viewer/pull/4643#discussion_r1463845714

https://model-viewer-xr-drag.web.app/unfixed/ — it uses the code from this pull request (without the hack described in the https://github.com/google/model-viewer/pull/4643#discussion_r1463845714)

https://model-viewer-xr-drag.web.app/original/ — it uses the original code

Just point the ray at the 3D model and press and hold the trigger on the controller to select and rotate the model as you want using the controller.

@elalish Please check them too. :wink:

Reply:
I tried out the demo. Very cool! The fixed version is already a big improvement for viewing on a Meta Quest.

Reply:
@FluorescentHallucinogen  👍  great job so far.

Reply:
I tried your solution with other models, the "drag and drop" doesn't seem to work for all GLBs.

Reply:
`return raycaster.intersectObjects(group.children, **true**);`
        
This should be set to **true**

Reply:
@sprengerst But why? From what I can see, it's `true` by default. See https://threejs.org/docs/#api/en/core/Raycaster.intersectObjects.

Reply:
I fixed this by adding a pivot so that we no longer set the position/orientation of the top-level scene object.